iidm/iidm-api/src/main/java/com/powsybl/iidm/network/BranchAdder.java

```diff
@@ -10,22 +10,23 @@
  *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
-public interface BranchAdder<T extends BranchAdder> extends IdentifiableAdder<T> {
+public interface BranchAdder<T extends Branch<? super T>, A extends BranchAdder> extends IdentifiableAdder<T, A> {
 
-    T setVoltageLevel1(String voltageLevelId1);
+    A setVoltageLevel1(String voltageLevelId1);
 
-    T setNode1(int node1);
+    A setNode1(int node1);
 
-    T setBus1(String bus1);
+    A setBus1(String bus1);
 
-    T setConnectableBus1(String connectableBus1);
+    A setConnectableBus1(String connectableBus1);
 
-    T setVoltageLevel2(String voltageLevelId2);
+    A setVoltageLevel2(String voltageLevelId2);
 
-    T setNode2(int node2);
+    A setNode2(int node2);
 
-    T setBus2(String bus2);
+    A setBus2(String bus2);
 
-    T setConnectableBus2(String connectableBus2);
+    A setConnectableBus2(String connectableBus2);
 
+    T add();
```

Comment:
I supposed that you added it here because of the generic class, but can you add `@Override`?